### ISSUE TITLE
Job hangs if repository that sent merge/pull request is private

### ISSUE THREAD

**Issue by mojavelinux** [appveyor/ci#2791]
If the target repository is public (and configured as such in AppVeyor), but the fork is private, AppVeyor will hang indefinitely trying to fetch from the private repository. I would expect AppVeyor to fail with an error message.

This can happen if the repository is initial private, someone forks it during that time, then the origin is changed to public. The fork will still be private.

**Comment by IlyaFinkelshteyn** [comment]
Right, @Ede123 found that too: https://github.com/appveyor/ci/issues/651#issuecomment-427102312. This is the reason we keep main issue open. We will probably skip build but right a descriptive event (project `Events` tab) in this case.
